src/opnsense/scripts/unbound/start.sh

```diff
@@ -30,7 +30,25 @@ set -e
 
 # prepare and startup unbound, so we can easily background it
 
-chroot -u unbound -g unbound / /usr/local/sbin/unbound-anchor -a /var/unbound/root.key
+# if the root.key file is missing or damaged, run unbound-anchor
+if ! /usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null; then
+	# unbound-anchor has undefined behaviour if file is corrupted, start clean 
+	rm -f /var/unbound/root.key
+
+	# if we are in forwarding mode, prefer to use the configured system nameservers
+	if [ -n "$(/usr/local/sbin/pluginctl -g unbound.forwarding)" ]; then
+	  if [ ! -f /var/unbound/resolv.conf.root ]; then
```

Comment:
maybe we could just check if the file is non empty, in that case use it avoiding to use pluginctl -g here (of course only filling the file with contents on this setting.